diff --git a/libr/util/thread.c b/libr/util/thread.c
index 309b8e3..5821ca6 100644
--- a/libr/util/thread.c
+++ b/libr/util/thread.c
@@ -149,7 +149,7 @@ R_API bool r_th_setaffinity(RThread *th, int cpuid) {
 	CPU_ZERO(&c);
 	CPU_SET(cpuid, &c);
 
-	if (sched_setaffinity (th->tid, sizeof (c), &c) != 0) {
+	if (sched_setaffinity ((pid_t)th->tid, sizeof (c), &c) != 0) {
 		R_LOG_ERROR ("Failed to set cpu affinity");
 		return false;
 	}
